First pass at mission surveys

igorkofman opened the pull request:
Please leave feedback!!

Commit:
Center background earth image.

igor47 committed:
rename `celery-worker` to `worker`

allows switching worker queues without renaming stuff

igor47 committed:
replace celery with tasktiger

igor47 committed:
email: don't silently fail on send errors

we might want to capture specific exceptions in specific situations --
like, in test if we are unauthorized

igor47 committed:
better log config based on root logger

so we're not limiting our logs to the `spaceship` space

igor47 committed:
fix tasktiger logging

for some reason, doing this prevents it from logging debug-level
messages every second.

jeredw committed:
set environment for workers

jeredw committed:
don't add a newline

igorkofman committed:
First pass at mission surveys

igor47 committed:
dockerignore: sendgrid key

this is just for local use and shouldn't end up in containers

igor47 committed:
use our custom fork of task tiger

igor47 committed:
Make our FROM address say "spaceship earth"

also adds our logo to the confirmation email

igor47 commented on migrations/versions/c28413616223_add_survey_answers_table.py:
```diff
@@ -0,0 +1,40 @@
+    sa.Column('id', sa.Integer(), autoincrement=True, nullable=False),
+    sa.Column('mission_id', sa.Integer(), nullable=False),
+    sa.Column('user_id', sa.Integer(), nullable=False),
+    sa.Column('survey_version', sa.Integer(), nullable=True),
```
think instead of survey version and question id, we should just store the question in text?

igor47 commented on spaceship/templates/eval.html:
```diff
@@ -0,0 +1,14 @@
+{% extends "base.html" %} {% block title %} Mission Debrief - {{ super() }} {%
```
weird text wrapping here, and in a few other `html` templates in this PR. maybe turn off the wrap feature?

igor47 commented on spaceship/views.py:
```diff
@@ -174,7 +212,7 @@ def create_mission(team_id):
 
     # schedule emails
-    email.schedule_mission_emails(mission)
+    #email.schedule_mission_emails(mission)
```
did you mean to comment this out?

igor47 commented on spaceship/views.py:
```diff
@@ -144,7 +145,44 @@ def cancel_mission_ajax(mission_id):
+@app.route('/mission/<mission_id>/report', methods=['GET', 'POST'])
+def report(mission_id):
+  mission = Mission.query.filter(Mission.id == mission_id).first()
+  team_ids = map(lambda t: t.id, current_user.teams)
```
lol this is javascript. you want:

```python
team_ids = [t.id for t in current_user.teams]
```

igor47 commented on spaceship/views.py:
```diff
@@ -144,7 +145,44 @@ def cancel_mission_ajax(mission_id):
+    return redirect(url_for('report', mission_id=mission_id))
+  else:
+    return render_template('eval.html', form=survey, mission=mission)
+{}
```
what's up with these curly braces? should this just be a newline?

igor47 reviewed (approved):
looks good, but you should uncomment sending mission emails. left a few other random comments

igorkofman committed:
Make it possible to join multiple times through a link

igor47 committed:
turn invite modal into a vue component

we have a JS function that populates the correct information. this makes
it possible/easier to reuse the modal for other emails

igor47 committed:
emailSender: preserve state

internally saves edits to messages, so if you're switching between
several senders the modal still contains your edits when you close and
re-open it

igor47 committed:
emailSender: handle actual send from vue app

igor47 committed:
sendEmail: rename modal

igor47 committed:
add resource integrity for additional loaded JS

might as well use it consistently. i used this site to generate the
hashes: https://www.srihash.org/
